R/format_message.R

```diff
@@ -81,6 +89,59 @@ format_message <- function(string, ..., line_length = 0.9 * options()$width, ind
   string
 }
 
+#' @name format_alert
+#' @rdname format_message
+#'
+#' @param type Type of exception alert to raise.
+#'   Can be `"message"` for `message()`, `"warning"` for `warning()`,
+#'   or `"error"` for `stop()`.
+#' @param call. Logical. Indicating if the call should be included in the the
+#'   error message. This is usually confusing for users when the function
+#'   producing the warning or error is deep within another function, so the
+#'   default is `FALSE`.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   format_alert("This is a message.")
+#'   format_alert("This is a warning.", type = "warning")
+#'   format_warning("This is a warning.")
+#'   format_error("This is an error.")
+#' }
+format_alert <- function(string, ..., line_length = 0.9 * getOption("width", 89), indent = "  ", type = c("message", "warning", "error"), call. = FALSE) {
+  type <- match.arg(type, choices = c("message", "warning", "error"))
+  if (type == "message") {
+    message(format_message(
+      string = string, ...,
+      line_length = line_length, indent = indent
+    ))
+  } else if (type == "warning") {
+    warning(format_message(
+      string = string, ...,
+      line_length = line_length, indent = indent
+    ), call. = call.)
+  } else {
+    stop(format_message(
+      string = string, ...,
+      line_length = line_length, indent = indent
+    ), call. = call.)
+  }
+}
+
+#' @name format_warning
+#' @rdname format_message
+#' @export
+format_warning <- function(..., type = "warning") {
```

Comment:
I would probably even hardcode here, i.e. no `type` argument and then call `format_alert(..., type = "warning")`. Same for `format_error()`.